MapboxVision/VisionManager/VisionReplayManager.swift

```diff
@@ -43,29 +43,45 @@ public final class VisionReplayManager: BaseVisionManager {
      - Returns: Instance of `VisionReplayManager` configured to use data from specified session.
      */
     public static func create(recordPath: String) throws -> VisionReplayManager {
-        return VisionReplayManager(dependencies: try ReplayDependencies.default(recordPath: recordPath))
+        VisionReplayManager(dependencies: try ReplayDependencies.default(recordPath: recordPath))
     }
 
     /**
      Video source that provides frames from recorded video.
      */
     public var videoSource: VideoSource {
-        return dependencies.player
+        dependencies.player
+    }
+
+    /// Duration of the session in seconds
+    public var duration: Float {
+        dependencies.native.duration
+    }
+
+    /// Current progress of the session in seconds
+    public var progress: Float {
+        get {
+            dependencies.native.progress
+        }
+        set {
+            dependencies.native.progress = newValue
+        }
     }
 
     /**
      Start delivering events from `VisionReplayManager`.
-     Calling `start` on already started or destroyed instance is considered a mistake.
+     Also starts playing the video. If the playing was stopped, then it will resume from progress at the momment of stopping.
+     Calling `start` on destroyed instance is considered a mistake.
 
-     - Important: Do NOT call this method more than once or after `destroy` is called.
+     - Important: Do NOT call this method after `destroy` is called.
      */
     public func start() {
         switch state {
         case .uninitialized:
             assertionFailure("VisionManager should be initialized before starting")
             return
         case .started:
-            assertionFailure("VisionManager is already started")
+            resume()
```

Comment:
because the methods probably will be used in ui elements therefore they should be tolerant to multiple touches, I suppose you don't want to get assertion failure because of two touches on pause button. Of course the problem can be solved on user's side, but in that case the user will need to implement extra logic for watching the replaying state

Reply:
When methods were designed we decided to provide asserts on subsequent calls of the same methods to allow developers to catch errors early.

Reply:
the logic of these methods were redesigned for replay, now they are more `pause`/`resume` than `stop`/`start`. Previously the methods wasn't supposed to be triggered by the end-users, now they may have such option

Reply:
The logic of this method should be exactly the same as for `VisionManager`. Nothing really changed in terms of availability for end-users: a developer is responsible for such exposure. We are providing early notification if these methods are used incorrectly.